spec/safe_statements_spec.rb

```diff
@@ -1263,6 +1281,239 @@ def up
           end
         end
 
+        describe "safe_add_concurrent_partitioned_index" do
+          before do
+            ActiveRecord::Base.connection.execute(<<~SQL)
+              CREATE SCHEMA partman;
+              CREATE EXTENSION pg_partman SCHEMA partman;
+            SQL
+          end
+
+          it "creates valid index when there are no child partitions" do
+            create_range_partitioned_table(:foos3, migration_klass)
+
+            test_migration = Class.new(migration_klass) do
+              def up
+                safe_add_concurrent_partitioned_index :foos3, :updated_at
+              end
+            end
+
+            allow(ActiveRecord::Base.connection).to receive(:execute).and_call_original
+
+            test_migration.suppress_messages { test_migration.migrate(:up) }
+
+            expect(ActiveRecord::Base.connection).to have_received(:execute).with(/CREATE INDEX/).once
```

Comment:
Personal preference: could we use the `to receive` style (not the `to have_received` style)?

Reply:
So I _tried_ using `to receive` but I think rspec gets confused when you have multiple expectations on the same method with different arguments.

Reply:
Fixed in https://github.com/braintree/pg_ha_migrations/pull/85/commits/a473f1fb102e8882d1ba0fbc11cb038e5e3bcb5a